src/Sulu/Bundle/AdminBundle/Resources/js/stores/userStore/userStore.js

```diff
@@ -142,10 +142,12 @@ class UserStore {
         }
 
         if (this.user) {
-            // when the user was logged in already and comes again with the same user
-            // we don't need to initialize again
+            // When the user was logged in already and comes again with the same user
+            // we don't need to initialize again.
+            // But after 2FA verification data.username is undefined, so at this point
+            // we do not know who the new user is. So we have to initialize nevertheless.
 
-            if (data.username === this.user.username) {
+            if (data.username && data.username === this.user.username) {
```

Comment:
I find this a little bit confusing as if data.username is undefined then `if (data.username === this.user.username) {` should not match.

So is it maybe more that `this.user` is already undefined that would be in my case a more logical if statement if we go this way:

```suggestion
            if (this.user && this.user.username === this.user.username) {
```

Can you check if that works?

Reply:
`this.user` is defined. It is an object, but it does not have a `username` property. Here's a screenshot of the debugger:
![grafik](https://github.com/user-attachments/assets/82660f1c-e488-4c79-b260-e3e2c8c5f6da)

Reply:
Looks like the `/admin/config` route does return this user data without the username. I'll try to find out, why this happens.

Reply:
Sounds strange is that only happening on SSO?

Reply:
I am not using SSO. Or did you mean 2FA?

Reply:
So I think to be safe we need then:

```js
if (this.user.username && this.user.username === data.username) {
```

Reply:
I have changed the condition, now and added the serialization group 'frontend' to the `username` property.